src/components/virtualSessions/VirtualSession.tsx

```diff
@@ -0,0 +1,202 @@
+  font-family: "Lato", sans-serif;
+  color: #212529;
+  text-align: center;
+`
+
+const Title = styled.div`
+  font-size: 28px;
+  text-align: center;
+`
+
+const TimePreferences = styled.div`
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: space-evenly;
+  text-align: center;
+  padding: 40px 0 40px 0;
+`
+
+const TimePreference = styled.div`
+  padding: 20px 0 20px 0;
+`
+
+const TimeWeekday = styled.div`
+  font-size: 26px;
+  color: #212529;
+`
+
+const MonthDay = styled.div`
+  font-size: 26px;
+  color: #212529;
+`
+
+const ConfirmSection = styled.div`
+  display: flex;
+  justify-content: center;
+`
+
+const ConfirmButton = styled.button`
+  background-color: #FCC600;
+  border: none;
+  border-radius: 4px;
+  color: #000000;
+  padding: 15px 62px;
+  text-align: center;
+  text-decoration: none;
+  display: inline-block;
+  font-size: 16px;
+  margin: 4px 2px;
+
+  :disabled {
+    border: 1px solid #999999;
+    background-color: #cccccc;
+    color: #666666;
+  }
+`
+
+interface ActionButtonProps {
+  active: boolean;
+}
+
+const div: StyledFunction<
+  ActionButtonProps & React.HTMLProps<HTMLInputElement>
+> = styled.div;
+
+const ActionButton = div`
+  font-family: 'Lato', sans-serif;
+  border: none;
+  border-radius: 4px;
+  padding: 15px 32px;
+  text-align: center;
+  text-decoration: none;
+  display: inline-block;
+  font-size: 16px;
+  cursor: pointer;
+  background-color: ${(props: ActionButtonProps) =>
+    props.active ? "#fcc600" : "#cccccc"};
+  color: ${(props: ActionButtonProps) =>
+    props.active ? "#000000" : "#666666"};
+`;
+
+interface IVirtualSessionParams {
+  id: string;
+}
+
+interface IVirtualSessionState {
+  virtualSession?: VirtualSessionInterface;
+  timePreferences?: string[];
+  preferenceSelected: number;
+  virtualSessionValid: boolean;
+  actionButtonText: string;
+}
+
+export class VirtualSession extends React.Component<RouteComponentProps<IVirtualSessionParams>, IVirtualSessionState> {
+  public apiService = new ServiceResolver().ApiService();
+  constructor(props: RouteComponentProps<IVirtualSessionParams>) {
+    super(props);
+    this.state = {
+      virtualSession: undefined,
+      timePreferences: undefined,
+      preferenceSelected: 0,
+      virtualSessionValid: true,
+      actionButtonText: 'Confirm'
+    }
+  }
+
+  public async componentDidMount() {
+    const virtualSession = await this.apiService.getVirtualSession(this.props.match.params.id);
+    const virtualSessionValid = virtualSession.volunteerId === "00000000-0000-0000-0000-000000000000";
+    const timePreferences = [virtualSession.timePreferenceOne!, virtualSession.timePreferenceTwo!, virtualSession.timePreferenceThree!];
+    this.setState({
+      virtualSession,
+      timePreferences,
+      virtualSessionValid
+    });
+  }
+
+  public handleTimePreferenceSelect = (preferenceSelected: number) => {
+    this.setState({
+      preferenceSelected
+    });
+  }
+
+  public handleConfirmClick = async () => {
```

Comment:
Also here we probably want to add a confirmation alert saying the confirmation was accepted or submitted successfully.

Reply:
cool, agreed. i'll make the change.